### PR TITLE
feat: use Weasel instead of spaCy everywhere

### PR DESCRIPTION
**Pull request by bdura**
This PR changes docstrings and messages to display Weasel instead of spaCy projects.

### CODE REVIEW

File: weasel/cli/assets.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ def project_assets(
         checksum = asset.get("checksum")
         if "git" in asset:
             git_err = (
-                "Cloning spaCy project templates requires Git and the 'git' command. "
+                "Cloning Weasel project templates requires Git and the 'git' command. "
```
Comment (rmitsch) [inline weasel/cli/assets.py:81]: What's the official spelling, "Weasel" or "weasel"?

Comment (bdura) [reply weasel/cli/assets.py:81]: Good point. I used Weasel/`weasel` v. spaCy/`spacy` but maybe that's not what @svlandeg had in mind?

Comment (bdura) [reply weasel/cli/assets.py:81]: (the README uses weasel, so I can make an educated guess)

Comment (svlandeg) [reply weasel/cli/assets.py:81]: I'm happy with Weasel/`weasel` !

File: weasel/cli/document.py [hunk 2]
```diff
@@ -8,23 +8,23 @@
 DOCS_URL = "https://spacy.io"
```
Comment (svlandeg) [inline weasel/cli/document.py:8]: Needs to be adjusted as well

File: weasel/cli/document.py [hunk 3]
```diff
@@ -8,23 +8,23 @@
...
 INTRO_PROJECT = f"""The [`{PROJECT_FILE}`]({PROJECT_FILE}) defines the data assets required by the
 project, as well as the available commands and workflows. For details, see the
-[spaCy projects documentation]({DOCS_URL}/usage/projects)."""
+[Weasel documentation]({DOCS_URL}/usage/projects)."""
 INTRO_COMMANDS = f"""The following commands are defined by the project. They
-can be executed using [`spacy project run [name]`]({DOCS_URL}/api/cli#project-run).
+can be executed using [`weasel run [name]`]({DOCS_URL}/api/cli#project-run).
```
Comment (svlandeg) [inline weasel/cli/document.py:13]: Everything related to the documentation needs to be revisited. If we say "Weasel documentation" than the docs need to move to this repo, and we need to make sure the links used here all work. Maybe it's best to address this specific bit in a separate PR?

File: weasel/cli/dvc.py [hunk 4]
```diff
@@ -1,4 +1,4 @@
-"""This module contains helpers and subcommands for integrating spaCy projects
+"""This module contains helpers and subcommands for integrating Weasel
 with Data Version Controk (DVC). https://dvc.org"""
```
Comment (svlandeg) [inline weasel/cli/dvc.py:2]:
```suggestion
with Data Version Control (DVC). https://dvc.org"""
```
This typo just caught my eye...

File: weasel/cli/document.py (hunk 3 above)
Comment (bdura) [reply weasel/cli/document.py:13]: I agree referring to a Weasel documentation and then adding a link to spaCy is a bit baroque... Let's leave this as is and move all the documentation in another PR then?

**Pull request merged**